### PR TITLE
Restrict ENI usage to IPv4

### PR DESCRIPTION
**Pull request by tgraf**
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/8843)

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

File: pkg/option/config.go [hunk 1]
```diff
@@ -1300,6 +1300,10 @@ func (c *DaemonConfig) Validate() error {
 		return fmt.Errorf("MTU '%d' cannot be negative", c.MTU)
 	}
 
+	if c.IPAM == IPAMENI && c.EnableIPv6 {
```
Comment (ianvernon) [inline pkg/option/config.go:1303]: Is this a limitation on the AWS side? A comment linking to according documentation on this restriction would be helpful.

Comment (tgraf) [reply pkg/option/config.go:1303]: AWS does support IPv6 but it requires manual steps. However, Cilium currently doesn't support it. I'll add a note to the ENI guide.

Comment (tgraf) [reply pkg/option/config.go:1303]: Documented.

**Review by ianvernon — APPROVED** [review]

**Comment by tgraf** [comment]
test-me-please

**Pull request merged**